shared/core/types/pools.go

```diff
@@ -26,6 +26,15 @@ func init() {
 	}
 }
 
+func PoolAddressToFriendlyName(address string) string {
```

Comment:
Rather than having a loop and checking this every time, I suggest adding a map (similar to the approach above) that only gets compiled once.